core/embed/rust/src/ui/model_tt/layout.rs

```diff
@@ -440,42 +440,98 @@ extern "C" fn new_confirm_action(n_args: usize, args: *const Obj, kwargs: *mut M
     verb: Option<StrBuffer>,
     verb_cancel: Option<StrBuffer>,
+    info_button: bool,
     hold: bool,
-) -> Result<Obj, Error> {
-    let paragraphs = ConfirmBlob {
-        description: description.unwrap_or_else(StrBuffer::empty),
-        extra: extra.unwrap_or_else(StrBuffer::empty),
-        data: data.try_into()?,
-        description_font: &theme::TEXT_NORMAL,
-        extra_font: &theme::TEXT_DEMIBOLD,
-        data_font: &theme::TEXT_MONO,
-    }
-    .into_paragraphs();
+}
 
-    let obj = if hold {
-        LayoutObj::new(Frame::left_aligned(
-            theme::label_title(),
-            title,
-            SwipeHoldPage::new(paragraphs, theme::BG),
-        ))?
-    } else if let Some(verb) = verb {
-        let buttons = Button::cancel_confirm_text(verb_cancel, Some(verb));
-        LayoutObj::new(Frame::left_aligned(
-            theme::label_title(),
+impl ConfirmBlobParams {
+    fn new(
+        title: StrBuffer,
+        data: Obj,
+        description: Option<StrBuffer>,
+        verb: Option<StrBuffer>,
+        verb_cancel: Option<StrBuffer>,
+        hold: bool,
+    ) -> Self {
+        Self {
             title,
-            SwipePage::new(paragraphs, buttons, theme::BG).with_cancel_on_first_page(),
-        ))?
-    } else {
-        panic!("Either `hold=true` or `verb=Some(StrBuffer)` must be specified");
-    };
-    Ok(obj.into())
+            subtitle: None,
+            data,
+            description,
+            extra: None,
+            verb,
+            verb_cancel,
+            info_button: false,
+            hold,
+        }
+    }
+
+    fn with_extra(mut self, extra: Option<StrBuffer>) -> Self {
+        self.extra = extra;
+        self
+    }
+
+    fn with_subtitle(mut self, subtitle: Option<StrBuffer>) -> Self {
+        self.subtitle = subtitle;
+        self
+    }
+
+    fn with_info_button(mut self, info_button: bool) -> Self {
+        self.info_button = info_button;
+        self
+    }
+
+    fn into_layout(self) -> Result<Obj, Error> {
+        let paragraphs = ConfirmBlob {
+            description: self.description.unwrap_or_else(StrBuffer::empty),
+            extra: self.extra.unwrap_or_else(StrBuffer::empty),
+            data: self.data.try_into()?,
+            description_font: &theme::TEXT_NORMAL,
+            extra_font: &theme::TEXT_DEMIBOLD,
+            data_font: &theme::TEXT_MONO,
+        }
+        .into_paragraphs();
+
+        let obj = if self.hold {
+            let mut frame = Frame::left_aligned(
+                theme::label_title(),
+                self.title,
+                SwipeHoldPage::new(paragraphs, theme::BG),
+            );
+            if let Some(subtitle) = self.subtitle {
+                frame = frame.with_subtitle(theme::label_subtitle(), subtitle);
+            }
+            if self.info_button {
+                frame = frame.with_info_button();
+            }
+            LayoutObj::new(frame)?
+        } else if let Some(verb) = self.verb {
```

Comment:
From 80 %, this `if` and `else if` branches are the same, I wonder if we can improve it.

Rough idea:
```rs
if !self.hold | self.verb.is_none() {
    panic!("Either `hold=true` or `verb=Some(StrBuffer)` must be specified");
}
let content = if self.hold {
    SwipeHoldPage::new(paragraphs, theme::BG)
} else {
    let buttons = Button::cancel_confirm_text(self.verb_cancel, Some(verb));
    SwipePage::new(paragraphs, buttons, theme::BG).with_cancel_on_first_page()
}
let mut frame = Frame::left_aligned(
    theme::label_title(),
    self.title,
    content,
);
if let Some(subtitle) = self.subtitle {
    frame = frame.with_subtitle(theme::label_subtitle(), subtitle);
}
if self.info_button {
    frame = frame.with_info_button();
}
let obj = LayoutObj::new(frame)?;
Ok(obj.into())
```

I am realising this will most probably not compile, because `SwipeHoldPage` and `SwipePage` are different types, but I believe it should be solvable somehow